tests/state/ViewerState/vector_slice.test.ts

```diff
@@ -3,8 +3,48 @@ import { Color } from "three";
 import { describe, expect, it } from "vitest";
 
 import { VECTOR_KEY_MOTION_DELTA, VectorTooltipMode } from "../../../src/colorizer";
+import { UrlParam } from "../../../src/colorizer/utils/url_utils";
 import { useViewerStateStore } from "../../../src/state";
+import { loadVectorSliceFromParams, serializeVectorSlice, VectorSlice } from "../../../src/state/slices";
+import { SerializedStoreData } from "../../../src/state/types";
 import { ANY_ERROR } from "../../test_utils";
+import { compareSerializedData, compareSlice } from "./utils";
+
+const EXAMPLE_SLICE_1: Partial<VectorSlice> = {
+  vectorVisible: true,
+  vectorKey: VECTOR_KEY_MOTION_DELTA,
+  vectorMotionTimeIntervals: 1,
+  vectorColor: new Color(0xff0000),
+  vectorScaleFactor: 2,
+  vectorTooltipMode: VectorTooltipMode.COMPONENTS,
+};
+
+const EXAMPLE_SLICE_1_PARAMS: SerializedStoreData = {
+  [UrlParam.SHOW_VECTOR]: "1",
+  [UrlParam.VECTOR_KEY]: VECTOR_KEY_MOTION_DELTA,
+  [UrlParam.VECTOR_TIME_INTERVALS]: "1",
+  [UrlParam.VECTOR_COLOR]: "ff0000",
+  [UrlParam.VECTOR_SCALE]: "2",
+  [UrlParam.VECTOR_TOOLTIP_MODE]: VectorTooltipMode.COMPONENTS,
+};
+
```

Comment:
how same/different are these from your other examples? maybe move them into a shared test-data file

Reply:
Hmm, these are scoped specifically to the vector slice so I'm not sure if it makes sense to make a big global state? In #596 I do add a big integration test across all the slices in `tests/state/store_io.test.ts` which contains something similar, but it's more of a gut check since it doesn't test multiple different values like this test does.